contracts/2key/singleton-contracts/TwoKeyUpgradableExchange.sol

```diff
@@ -204,6 +238,30 @@ contract TwoKeyUpgradableExchange is Upgradeable, MaintainingPattern {
         return tokens;
     }
 
+
+    function swapEthForStableCoin(uint ethWeiAmount) internal returns (uint){
+        uint256 minConversionRate;
+        uint256 stableCoinUnits;
+        IKyberNetworkProxy proxyContract;
+        proxyContract = IKyberNetworkProxy(kyberProxyContractAddress);
+        (minConversionRate,) = proxyContract.getExpectedRate(ETH_TOKEN_ADDRESS, daiAddress, 100000000000000000);
+        stableCoinUnits = proxyContract.swapEtherToToken.value(ethWeiAmount)(daiAddress,minConversionRate).div(10**18);
+        usdStableCoinUnitsReserve += stableCoinUnits;
+    }
+
+
+    function buyStableCoinWith2key(uint _twoKeyUnits, address _beneficiary) external onlyValidatedContracts returns (uint){
+        uint usdTetheredStableCoinUnits;
+
+        usdTetheredStableCoinUnits = _getUsdStableCoinAmountFrom2keyUnits(_twoKeyUnits, twoKeyToStableCoinExchangeRate);
+
+        require(usdStableCoinUnitsReserve - usdTetheredStableCoinUnits > 0);
+
+        usdStableCoinUnitsReserve -= usdTetheredStableCoinUnits;
```

Comment:
in case we've been using safe math it would be enough to do usdStableCoinUnitsReserve.sub(usdTetheredStableCoinUnits)